### PR TITLE
Filter Service Participants that show up on Track Attendance by status, role and stage

### PR DESCRIPTION
**Pull request by ptewson-sfdo**
This work adds design properties to the attendance tracker component (visible on the service session page) to allow admins to filter untracked participants by the Status of Service Participant and Role or Stage of the Program Engagement related to the record. If no filters are specified (what customers will experience at the time this feature is released) than no filtering is applied, meaning that the customer experience should be unchanged until they chose to apply filters. 

# Critical Changes

# Changes

# Issues Closed
- [W-9655326](https://gus.lightning.force.com/a07AH000000SBNjYAO)

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
- ~[] Any net new LWC work has Sa11y tests & 50% or above lines JEST test coverage~
- [X] CRUD/FLS is enforced in Apex
- ~[ ] Permission sets are updated to account for CRUD|FLS|Tab|Classes~
- ~[ ] Field sets are updated to account for new fields~
- [X] UX approval or UX not necessary
- [X] Link the pull request and work item by PR comment and Chatter post respectively, e.g. GUS: W-0000000: Work Name (https://github.com/SalesforceFoundation/PMM/pull/303)
- [ ] All **acceptance criteria** have been met
    - [X] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [ ] PR contains draft release notes
- [ ] QE story level testing completed

### CODE REVIEW

**Review by dospromptman — CHANGES_REQUESTED** [review]
One label in the component properties might need to be updated

**Comment by dospromptman** [comment]
I think I remember in the discussion of this work item during grooming that we said we wouldn't need to worry about commas existing in picklist values, but I tried adding a picklist value with a comma in both the value and api name and it accepted it.  I'm not sure this logic will support cases where users have added values like that and are trying to filter them using the component properties.

File: force-app/main/default/lwc/attendance/attendance.js-meta.xml [hunk 1]
```diff
@@ -14,9 +14,57 @@
             <objects>
                 <object>ServiceSession__c</object>
             </objects>
+            <property name="omitServiceParticipantStatuses" 
+                      label="Excluded Service Participant Statuses"
```
[the remark this answers is not in the document]
Comment (ptewson-sfdo) [reply force-app/main/default/lwc/attendance/attendance.js-meta.xml:18]: I believe so - I will confirm that works and then make that update, thanks!

File: force-app/main/default/lwc/attendance/attendance.js-meta.xml [hunk 2]
```diff
@@ -14,9 +14,57 @@
...
+                      default=""
+                      type="String" />
+            <property name="omitProgramEngagementRoles"
+                      label="Excluded Program Engagement Roles"
+                      description="A comma-separated list of the API values used to exclude participants from Attendance Tracker." 
+                      default=""
+                      type="String" />
+            <property name="omitProgramEngagementStages"
+                      label="Excluded Program Engagement Stages"
+                      description="A comma-delimeted list of API stage values to omit"
```
[the remark this answers is not in the document]
Comment (ptewson-sfdo) [reply force-app/main/default/lwc/attendance/attendance.js-meta.xml:64]: Sure should, good catch!

File: force-app/main/default/lwc/attendance/attendance.js-meta.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ptewson-sfdo) [reply force-app/main/default/lwc/attendance/attendance.js-meta.xml:18]: I'm not sure its possible to use labels for translation here - I definitely want to keep looking into it, but so far as I can tell the label support works when supplying the *value* of the design property, but so far I have yet to find success using custom labels to define the label or description within the meta.xml file. 

For instance, this works to apply the filter:
image


But attempting to use labels like so:
```
            <property name="omitProgramEngagementStages"
                      label="{!$Label.Incorrect_Tab}"
                      description="{!$Label.Incorrect_Tab}"
                      default=""
                      type="String" />   
```
Just renders like this:
image

**Review by dospromptman — APPROVED** [review]

**Pull request merged**